Makefile

```diff
@@ -4,6 +4,11 @@ all:
 	(cd rust; cargo build)
 	doxygen
 
+check:
+	$(MAKE) -C c-layer $@
+	$(MAKE) -C c-example $@
+	(cd rust; cargo test -- --test-threads=1)
```

Comment:
please document here why that single thread is needed. It is easier to spot it in code then in commit message/pull request or specific test.